File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def dfs(depth, numbers, target, su):
    if depth == len(numbers):
        if su == target:
            return 1
        else:
            return 0
    return dfs(depth+1, numbers, target, su + numbers[depth]) + dfs(depth+1, numbers, target, su - numbers[depth])
    

def solution(numbers, target):
    return dfs(0, numbers, target, 0)

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <string>
#include <vector>

using namespace std;

int dfs(int sum, int pos, vector<int> numbers, int target){
    if(pos == numbers.size()){
        if(sum == target){
            return 1;
        }
        return 0;
    }
    return dfs(sum-numbers[pos], pos+1, numbers, target) + dfs(sum+numbers[pos], pos+1, numbers, target);
}

int solution(vector<int> numbers, int target) {
    int answer = 0;
    return dfs(0, 0, numbers, target);
}